File: src/test/java/com/ddd/trainings/api/app/TrainingIdeaService_propose_test.java
package com.ddd.trainings.api.app;

import com.ddd.trainings.TrainingTests;
import com.ddd.trainings.idea.domain.IdeaId;
import com.ddd.trainings.proposal.domain.ReviewerFactory;
import org.junit.jupiter.api.Test;

class TrainingIdeaService_propose_test extends TrainingTests {

  @Test
  void proposeIdea_createsProposal_onValid() {
    //given
    IdeaId ideaId = trainingIdeaBuilder.inDb().getId();
    reviewerRepo.save(ReviewerFactory.create());

    //when
    service.proposeIdea(ideaId.getValue());

    //then
    // later...
  }

  @Test
  void proposeIdea_error_onMissingReviewers() {
    //given
    IdeaId ideaId = trainingIdeaBuilder.inDb().getId();

    //expect
    // later...
  }
}

File: src/Ddd.Tests/Trainings/Idea/App/TrainingIdeaService_propose_test.cs
using Xunit;
using Ddd.Trainings.Idea.Domain;
using Ddd.Trainings.Proposal.Domain;
using Ddd.Tests.Trainings.Tests;

namespace Ddd.Tests.Trainings.Idea.App;

public class TrainingIdeaService_propose_test : TrainingTests
{
  [Fact]
  public void ProposeIdea_createsProposal_onValid()
  {
    //given
    var ideaId = TrainingIdeaBuilder.WithName("Test Idea long enough").InDb().IdeaId;
    ReviewerRepo.Save(ReviewerFactory.Create("name", ["idea"]));

    //when
    Service.ProposeIdea(ideaId.GetValue());

    //then
    // TODO: Test if proposal was created and persisted
  }

  [Fact]
  public void ProposeIdea_error_onMissingReviewers()
  {
    //given
    var ideaId = TrainingIdeaBuilder.InDb().IdeaId;

    //expect
    // not yet...
  }
}